test/util/tektonresources.virtualfs.test.ts

```diff
@@ -0,0 +1,231 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as os from 'os'
+import * as fs from 'fs';
+import * as path from 'path';
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import * as sinon from 'sinon';
+import { TektonResourceVirtualFileSystemProvider, kubefsUri } from '../../src/util/tektonresources.virtualfs';
+import { Uri, workspace, window, commands, TextDocument, EndOfLine } from 'vscode';
+import { CliImpl } from '../../src/cli';
+import * as k8s from 'vscode-kubernetes-tools-api';
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+suite('TektonResourceVirtualFileSystemProvider', () => {
+    let sandbox: sinon.SinonSandbox;
+    let v1Stub: sinon.SinonStub;
+    let osStub: sinon.SinonStub;
+    let unlinkSyncStub: sinon.SinonStub<[fs.PathLike], void>;
+    let openTextStub: sinon.SinonStub<[{ language?: string; content?: string }?], Thenable<import('vscode').TextDocument>>;
```

Comment:
`Thenable<import('vscode').TextDocument>>` is not good, you already imported `TextDocument` use that.